### PR TITLE
Calibration

### PR DESCRIPTION
**Pull request by rivkabr**

### CODE REVIEW

**Comment by rivkabr** [comment]
[(5) Microphone calibration](https://trello.com/c/QVuEZLJe/3595-5-microphone-calibration)

File: src/cljs/audyx/calibration/devices.cljs [hunk 1]
```diff
@@ -85,12 +85,11 @@
...
       [:error error] [:error (show-error-to-user $modal error :get-from-3rdparty)])))
 
-(deftry get-microphone [$modal serial-number]
+(deftry get-microphone [serial-number]
   (go 
     (match (<! (http-get microphones/resource serial-number))
       [:ok {:calibration res}] [:ok res]
-      [:error (:or 404 "Not Found")] (<! (create-micro $modal serial-number))
-      [:error error] [:error (show-error-to-user $modal get-microphone error)])))
+      [:error error] [:error (dbg error)])))
```
Comment (rinama) [inline src/cljs/audyx/calibration/devices.cljs]: Remove `dbg`!!!!!!!!

File: src/cljs/audyx/audio/spectrum_analyser.cljs [hunk 2]
```diff
@@ -67,12 +68,28 @@
...
-(defn analyse-audio-file [buffer set-octaves-fn set-sum-all-fn set-request-fn on-ended-cb]
+(defn microphone-calibration-to-fft-array [calibration]
+  (->> 
+    (devices/calc-micro-calib calibration)
+    :all-fft
+    (sort-by first)
+    (map second)
+    clj->js))
+
+(defn get-microphone-calibration [serial-number]
```
Comment (rinama) [inline src/cljs/audyx/audio/spectrum_analyser.cljs]:
```suggestion
(defn fetch-microphone-calibration [serial-number]
```

File: src/cljs/audyx/audio/spectrum_analyser.cljs [hunk 3]
```diff
@@ -67,12 +68,28 @@
...
+  (go
+    (let [[status calibration] (<! (devices/get-microphone serial-number))]
+      (if (= status :ok)
```
Comment (rinama) [inline src/cljs/audyx/audio/spectrum_analyser.cljs]: `match`

File: src/cljs/audyx/audio/spectrum_analyser.cljs [hunk 4]
```diff
@@ -67,12 +68,28 @@
                 analyser (audio-context/create-analyser {:fftSize fft-size :smoothingTimeConstant smoothing-time})
                 octaves-band (map (partial octave-band fft-size analyser) all-octaves)]
             (audio-context/connect source-node analyser)
-            [:ok (read-input-values analyser set-octaves-fn set-sum-all-fn set-request-fn octaves-band fft-size)])
+            [:ok (read-input-values analyser nil set-octaves-fn set-request-fn octaves-band fft-size true)])
         [:error error] [:error error]))))
 
-(defn analyse-audio-file [buffer set-octaves-fn set-sum-all-fn set-request-fn on-ended-cb]
+(defn microphone-calibration-to-fft-array [calibration]
```
Comment (rinama) [inline src/cljs/audyx/audio/spectrum_analyser.cljs]:
```suggestion
(defn microphone-calibration->-fft-array [calibration]
```

**Pull request merged**